### ISSUE TITLE
Python Script: JSON to CSV

### ISSUE THREAD

**Issue by bhargavnova** [bhargavnova/python-helper-modules#33]
### Objective

Develop a Python script that converts JSON (JavaScript Object Notation) data to CSV (Comma Separated Values) format. This script will provide a convenient way to transform structured JSON data into a tabular format that can be easily opened and manipulated in spreadsheet software.

### Features

- **Convert JSON to CSV**: Implement functionality to read JSON data and export it as a CSV file.
- **Handle Nested JSON**: Ensure that the script can handle nested JSON structures and appropriately flatten them for CSV representation.
- **Configurable Mapping**: Allow users to specify custom mappings or select specific fields from the JSON data for inclusion in the CSV file.

### Example Usage

```python
import json_to_csv

# Convert JSON to CSV
json_to_csv.convert('input.json', 'output.csv')

# Convert JSON with custom mapping
mapping = {'Name': 'full_name', 'Email': 'contact.email', 'Phone': 'contact.phone'}
json_to_csv.convert('input.json', 'output.csv', field_mapping=mapping)
```
`Difficulty`: Intermediate

`Tags`: Python, JSON, CSV, Data Conversion

### Additional Information
- Ensure that the script provides informative error messages in case of invalid JSON input or other issues during the conversion process.
- Consider using libraries like json and csv in Python for handling JSON and CSV operations respectively.

### Contribution Guidelines
The updated guidelines can be found [here](https://github.com/bhargavnova/python-helper-modules#contributing-guidelines).

### Note:
- `folder_name`: json_to_csv
- `script_name`: json_to_csv

**Comment by akadeepesh** [comment]
let me try this one

**Comment by akadeepesh** [comment]
yes I did, here it is

{
  "Name": "Deepesh kumar",
  "Age": 19,
  "Email": "xyz@example.com",
  "Phone": "+911231231239",
  "Address": {
    "Street": "xyz",
    "City": "New Delhi",
    "State": "Delhi",
    "PostalCode": "12345"
  },
  "Orders": [
    {
      "OrderID": "1001",
      "Product": "Sticker",
      "Quantity": 5
    },
    {
      "OrderID": "1002",
      "Product": "Hacktober",
      "Quantity": 3
    }
  ]
}

**Comment by ananyapillai72** [comment]
would like to work on this

**Comment by bhargavnova** [comment]
Sure!

**Comment by onkar-kota** [comment]
I'm too working on this issue

**Comment by bhargavnova** [comment]
Hi @onkar-kota ,
@ananyapillai72 was already working on this issue, she has also submitted her script,
so if you can make another script that can also handle nested dictionaries, if you are interested in that, you can work on that part.

For example, @ananyapillai72 , script can handle the below structure,
 ```[
    {
        "name": "Test37",
        "status": "done",
        "slug": "test-375960",
        "date": "13-10-2023",
        "author": "unknown",
        "probability": "89%",
        "result": "70%",
        "final_status": "failed"
    },
    {
        "name": "Test38",
        "status": "pending",
        "slug": "test-385960",
        "date": "13-10-2023",
        "author": "unknown",
        "probability": "80%",
        "result": null,
        "final_status": null
    
    }
]
```
Now you can working script that can handle below structure,
```
[
    {
        "name": "Test37",
        "status": "done",
        "slug": "test-375960",
        "date": "13-10-2023",
        "author": "unknown",
        "probability": "89%",
        "result": "70%",
        "final_status": "failed",
        "connected": {
            "run_again": true,
            "next_test": "Test38",
            "next_test_status": "pending"
        }
    },
    {
        "name": "Test38",
        "status": "pending",
        "slug": "test-385960",
        "date": "13-10-2023",
        "author": "unknown",
        "probability": "80%",
        "result": null,
        "final_status": null
    
    }
]
```

**Comment by onkar-kota** [comment]
Ohk got the point, ill try to solve this!

**Comment by bhargavnova** [comment]
Hi @ananyapillai72,
Thank you for your contribution :),
you can check on other open issues, or you can also submit one of your own.

Happy Coding!

**Comment by onkar-kota** [comment]
Hey @bhargavnova should I push to json_to_csv existing folder or should I create a new folder?

**Comment by bhargavnova** [comment]
Hi @onkar-kota , let's keep it in the same folder, otherwise things will be hard to manage.
name your script, `json_to_csv_with_nested_dict.py`

**Comment by onkar-kota** [comment]
Hi @bhargavnova I have resolved the issue can you please check my PR #63

**Comment by bhargavnova** [comment]
Hi @onkar-kota,
Thank you for your contribution :),
you can check on other open issues, or you can also submit one of your own.

Happy Coding!
